ast.cpp

```diff
@@ -32,31 +32,76 @@ class AstNode {
 class StmtNode : public AstNode {};
 /// @note This is an abstract class.
 class ExprNode : public AstNode {};
+/// @note This is an abstract class.
```

Comment:
```suggestion

/// @note This is an abstract class.
```
Since `DeclNode` is no longer an empty abstract class, let's add a blank line (vertical whitespace) to separate it from the abstract classes conceptually.